0001-typedb/02-dev/03-match.md

```diff
@@ -0,0 +1,792 @@
+
+1. Find all `person` entities (`$p`) that have a `full-name` attribute whose value is `Kevin Morrison`.
+2. Find all `object` entities (`$o`) that have a `path attribute` (`$o-path`).
+3. Find all `access` relations (`$ac`) where `$o` plays the `accessed-object` role.
+4. Find all `permission` relations (`$pe`) where `$p` plays the `permitted-subject` role, `$ac` the `permitted-access` role.
+
+The `object` entities are limited to those which play the `accessed-object` role in `access` relations, themselves 
+limited to those which play the `permitted-access` role in `permission` relations, themselves limited to those in 
+which matching `person` entities plays the `permitted-subject` role.
+
+#### Disjunctions
+
+To include statements in the form of a disjunction, we need to wrap each statement in `{}` and place the `or` keyword 
+in between them.
+
+<!-- test-ignore -->
+```typeql
+match
+  $p isa person, has full-name $n;
+  { $n contains "Masako"; } or { $n contains "Kevin"; };
+get $p;
+```
+
+The above query uses disjunctions to ensure one of two statements are matched:
+
+1. Finds all `person` entities that have a `full-name` attribute ($n).
+2. Checks to see if `$n` contains the text “Masako” **OR** if `$n` contains the text “Kevin”:
+   - If either statement is true, there is a match.
+   - If neither statement is true, there is NO match.
+
+The `person` entities are limited to those that have a `full-name` attribute whose value contains `Masako` or `Kevin`.
+
+#### Complex example
+
+To better illustrate the possibilities, we will now look at an example of a more complex pattern.
+
+![Complex example](../../images/query/example-pattern.png)
+
+The pattern is a conjunction of five different pattern types:
+
+1. **Conjunction 1** specifies the variables for two `person` instances, their `full-names`, `action` and `file` that 
+   has path `README.md`, specifies their types.
+2. **Disjunction** specifies that the actions of interest are either `modify_file` or `view_file`.
+3. **Negation 1** specifies that person `$p1` shall not have `full-name` with value of `Masako Holley`.
+4. **Negation 2** specifies that person `$p2` shall not have `full-name` with value of `Masako Holley`.
+5. **Conjunction 2** defines the pattern requiring the `file` to have `access` with `action` that we specified earlier, 
+   and both instances of `person` to have a `permission` to the specified `access`.
+
+In short, the above example finds pairs of people who are both have permission to access the same file 
+with a path of `README.md`. The pattern additionally specifies both of them to not have name `Masako Holley` and
+the access to be either `modify_file` or `view_file`.
+
+## Query examples
+
+### How to send a query
+
+The easiest way to send a query to the TypeDB server is to use [TypeDB Studio](../../02-clients/01-studio.md) to do so. 
+See the [Quickstart guide](../01-start/03-quickstart.md) for instructions on how to do this.
```

Comment:
"How to execute a query", or "how to query data" would be more in line with how other databases phrase this, from what I can find.